### PR TITLE
MAINT Fix CI linter check and fix linting issues

### PR DESCRIPTION
**Pull request by jeremiedbb**
all pre-commit hooks are always skipped on the CI. I guess it's because by default pre-commit only runs on staged changes but in the CI there are no staged changes. I propose to always run on all files (it's fast).

I added new rules for flake8 to ignore imports not at top of file for examples and confs.

### CODE REVIEW

**Comment by jeremiedbb** [comment]
commit [1feaefb](https://github.com/skrub-data/skrub/pull/664/commits/1feaefb98bf8661a269e5f90100fd33427215212) left 1 linting issue and the CI linter fails.

**Comment by jeremiedbb** [comment]
commit [90cfaef](https://github.com/skrub-data/skrub/pull/664/commits/90cfaeffc183739cd6d249b5f00360a4be4653b0) and commit [88cf1db](https://github.com/skrub-data/skrub/pull/664/commits/88cf1db57062c500455351d4ac9ad5455bb28cf3) fixed the last linting issues and no the CI linter is green !

**Comment by jeremiedbb** [comment]
Note that although `mypy` is listed in the dev dependencies and has some config flags in `setup.cfg`, it's not run by pre-commit and not checked in the CI. I'm not adding that in this PR because there's a huge pile of errors to fix :)

File: examples/04_fuzzy_joining_and_FeatureAugmenter.py [hunk 1]
```diff
@@ -385,12 +386,13 @@
 
 df_final.head(10)
 
+from sklearn.compose import make_column_transformer
+
 ##########################################################################
 # And that's it! As previously, we now have a big table
 # ready for machine learning.
 # Let's create our machine learning pipeline:
 from sklearn.pipeline import make_pipeline
```
Comment (LilianBoulard) [inline examples/04_fuzzy_joining_and_FeatureAugmenter.py:395]:
```suggestion
##########################################################################
# And that's it! As previously, we now have a big table
# ready for machine learning.
# Let's create our machine learning pipeline:
from sklearn.compose import make_column_transformer

from sklearn.pipeline import make_pipeline
```

**Review by LilianBoulard — APPROVED** [review]
Thanks a lot for the PR, that will be very useful!
Maybe we should add this PR to the [git-blame-ignore-revs](https://github.com/skrub-data/skrub/blob/main/.git-blame-ignore-revs) file?

**Review by LilianBoulard — APPROVED** [review]
Perfect, and thanks for fixing the CI changelog check :)

**Pull request merged**